hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/traceability/HederaTracer.java

```diff
@@ -133,9 +166,93 @@ private void trackNewAction(final MessageFrame messageFrame, final Consumer<Soli
     }
 
     private void finalizeActionFor(final SolidityAction action, final MessageFrame frame, final State frameState) {
+
+        switch (frameState) {
+            case NOT_STARTED, CODE_EXECUTING, CODE_SUSPENDED:
+                {
```

Comment:
This "extra-bracing" style appears to be out of style, but I cannot find any code style statements permitting or denying it.  Closest is https://google.github.io/styleguide/javaguide.html#s4.8.4-switch which shows a non-braced statement.

Reply:
Are you suggesting I'm not fashionable? Hmm. Actually, I'm less interested in what other random committees writing style guides think than I am in my own opinion of what is readable and what is not.  I happen to think it is more readable to have these blocks - with their very divergent handling of the cases - really separated in the reader's mind.

But of course I'll go with prevailing (Swirlds Labs) standards.

Reply:
"out of style" being the decreed Swirlds code style, which uses Google's as a default. I'm only arguing for consistency.  I honestly think "blocks" is a java language feature most don't know about.  The typical value added (creating variable scope and reducing name collisions) is not commonly needed.

Reply:
yes and sonar complains about blocks in some cases where they're nicely used to group statements that are logically very tightly connected but aren't strictly required.  but in this case it does _not_ complain.  and, as you pointed out, the google style guide says nothing about blocks used to group statements only (it only talks about blocks and "block-like constructs" in terms of indentation and such.  so ... what do you think?  ["whatever is not prohibited is allowed"](https://en.wikipedia.org/wiki/Everything_which_is_not_forbidden_is_allowed) - or vice versa?

Reply:
It will diff better if removed and is not strictly necessary.  The switch labels already nicely group it.  I'm of the opinion that it reduces readability and because it's a non-standard pattern it will distract readers on the style instead of focusing on the code.  I'm not aware of any other place in hedera code that uses free blocks.  While not strictly prohibited the code would be better if it was removed.  But this issue alone will not block the PR.